c61665245.lua

```diff
@@ -10,117 +10,72 @@ function c61665245.initial_effect(c)
-	end
-	return ok
-end
-function c61665245.sptg2(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
-	local c=e:GetHandler()
-	local zone={}
-	zone[0]=c:GetLinkedZone(0)
-	zone[1]=c:GetLinkedZone(1)
-	local lg=c:GetLinkedGroup()
-	if chkc then return chkc:IsLocation(LOCATION_MZONE) and chkc:IsControler(tp) and c61665245.spfilter_chkc(chkc,e,tp,lg,e:GetLabel()) end
-	if chk==0 then return Duel.IsExistingTarget(c61665245.spfilter2,tp,LOCATION_MZONE,LOCATION_MZONE,1,nil,e,tp,lg) end
-	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_TARGET)
-	local g=Duel.SelectTarget(tp,c61665245.spfilter2,tp,LOCATION_MZONE,LOCATION_MZONE,1,1,nil,e,tp,lg)
-	Duel.SetOperationInfo(0,CATEGORY_SPECIAL_SUMMON,nil,1,tp,LOCATION_DECK)
-	e:SetLabel(g:GetFirst():GetRace())
-end
-function c61665245.spop2(e,tp,eg,ep,ev,re,r,rp)
-	local c=e:GetHandler()
-	local tc=Duel.GetFirstTarget()
-	if not c:IsRelateToEffect(e) or not tc:IsRelateToEffect(e) or tc:IsFacedown() then return end
-	local zone={}
-	local flag={}
-	for p=0,1 do
-		zone[p]=c:GetLinkedZone(p)&0xff
-		local _,flag_tmp=Duel.GetLocationCount(p,LOCATION_MZONE,tp,LOCATION_REASON_TOFIELD,zone[p])
-		flag[p]=(~flag_tmp)&0x7f
+	local tc=Duel.SelectMatchingCard(tp,c61665245.spfilter1,tp,LOCATION_HAND,0,1,1,nil,e,tp,zone):GetFirst()
+	local function makeDisable(dc)
+		local e1=Effect.CreateEffect(c)
+		e1:SetType(EFFECT_TYPE_SINGLE)
+		e1:SetCode(EFFECT_DISABLE)
+		e1:SetReset(RESET_EVENT+RESETS_STANDARD)
+		dc:RegisterEffect(e1)
+		local e2=Effect.CreateEffect(c)
+		e2:SetType(EFFECT_TYPE_SINGLE)
+		e2:SetCode(EFFECT_DISABLE_EFFECT)
+		e2:SetValue(RESET_TURN_SET)
+		e2:SetReset(RESET_EVENT+RESETS_STANDARD)
+		dc:RegisterEffect(e2)
+		Duel.SpecialSummonComplete()
 	end
-	Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_SPSUMMON)
-	local g=Duel.SelectMatchingCard(tp,c61665245.spfilter3,tp,LOCATION_DECK,0,1,1,nil,e,tp,tc:GetRace())
-	local sc=g:GetFirst()
-	if sc then
-		local avail_zone=0
-		for p=0,1 do
-			if sc:IsCanBeSpecialSummoned(e,0,tp,false,false,POS_FACEUP_DEFENSE,p,zone[p]) then
-				avail_zone=avail_zone|(flag[p]<<(p==tp and 0 or 16))
+	if tc and Duel.SpecialSummonStep(tc,0,tp,1-tp,false,false,POS_FACEUP_DEFENSE,zone) then
+		makeDisable(tc)
+		local g=Duel.GetMatchingGroup(c61665245.spfilter2,tp,LOCATION_DECK,0,nil,e,tp,tc:GetRace())
+		if #g>0 and Duel.SelectYesNo(tp,aux.Stringid(61665245,1)) then
+			Duel.BreakEffect()
+			Duel.Hint(HINT_SELECTMSG,tp,HINTMSG_SPSUMMON)
+			local sc=g:Select(tp,1,1,nil):GetFirst()
+			if Duel.SpecialSummonStep(sc,0,tp,tp,false,false,POS_FACEUP) then
+				makeDisable(sc)
+				local e3=Effect.CreateEffect(c)
+				e3:SetType(EFFECT_TYPE_FIELD)
+				e3:SetCode(EFFECT_CANNOT_SPECIAL_SUMMON)
+				e3:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+				e3:SetTargetRange(1,0)
+				e3:SetTarget(c61665245.splimit)
+				e3:SetLabel(sc:GetOriginalRace())
+				e3:SetReset(RESET_PHASE+PHASE_END)
+				Duel.RegisterEffect(e3,tp)
```

Comment:
summon complete?

Reply:
summon complete is at the top function part